pkg/option/config.go

```diff
@@ -2840,7 +2840,7 @@ func (c *DaemonConfig) Populate(logger *slog.Logger, vp *viper.Viper) {
 	c.LocalRouterIPv6 = vp.GetString(LocalRouterIPv6)
 	c.EnableBPFClockProbe = vp.GetBool(EnableBPFClockProbe)
 	c.EnableIPMasqAgent = vp.GetBool(EnableIPMasqAgent)
-	c.EnableIPv4EgressGateway = vp.GetBool(EnableIPv4EgressGateway)
+	c.EnableEgressGateway = vp.GetBool(EnableEgressGateway)
```

Comment:
I like that we have just one config option in the daemon config, but we'll have to consolidate both the old (ipv4-only) and new feature flag. So if either `enable-egress-gateway` or `enable-ipv4-egress-gateway` is set, the daemon config option needs to be true.